### PR TITLE
feat: Add tool to compare gluon benchmark outputs

### PR DESCRIPTION
**Pull request by LBeernaertProton**

### CODE REVIEW

**Comment by LBeernaertProton** [comment]
example:
```
<bin> benchmark/report.json benchmark/report2.json
Benchmark imap-append: Fastest (90th Percentile=3.243148348s) benchmark/report2.json
Benchmark imap-fetch: Fastest (90th Percentile=859.002547ms) benchmark/report2.json
```

File: benchmarks/gluon_bench/tools/compare_bench_output.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+type JSONBenchmarkReport struct {
+	Name       string
+	Runs       []*JSONBenchmarkStatistics
+	Statistics *JSONBenchmarkStatistics
+}
+
+func loadReportFromFile(path string) ([]*JSONBenchmarkReport, error) {
+	contents, err := ioutil.ReadFile(path)
```
Comment (jameshoulahan) [inline benchmarks/gluon_bench/tools/compare_bench_output.go:32]: ioutil is deprecated, use `os.ReadFile` instead

File: benchmarks/gluon_bench/tools/compare_bench_output.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	// load reports
+	reportFiles := os.Args[1:]
+	reports := make([][]*JSONBenchmarkReport, 0, len(reportFiles))
+
+	for _, v := range reportFiles {
+		report, err := loadReportFromFile(v)
+		if err != nil {
+			panic(fmt.Errorf("failed to load report:%w", err))
```
Comment (jameshoulahan) [inline benchmarks/gluon_bench/tools/compare_bench_output.go:64]:
```suggestion
			panic(fmt.Errorf("failed to load report: %w", err))
```

Comment (LBeernaertProton) [reply benchmarks/gluon_bench/tools/compare_bench_output.go]: Done

File: benchmarks/gluon_bench/tools/compare_bench_output.go (hunk 1 above)
Comment (LBeernaertProton) [reply benchmarks/gluon_bench/tools/compare_bench_output.go]: Done

**Review by jameshoulahan — APPROVED** [review]

**Pull request merged**